File: 1624 B. Make AP/Solution.py
# 324906129
t = int(input())
while t>0:
    a,b,c = map(int,input().split())
    
    if (a+2*(b-a))>=c and (a+2*(b-a))%c==0 and (a+2*(b-a))!=0:
        print("YES")
        
    elif (2*b-c)>=a and (2*b-c)%a==0 and (2*b-c)!=0:
        print("YES")
        
    elif (a+(c-a)//2)>=b and (c-a)%2==0 and (a+(c-a)//2)%b==0 and (a+(c-a)//2)!=0:
        print("YES")
        
    else:
        print("NO")

    t -= 1

File: 1624 B. Make AP/Solution.c
// 324975548
#include<stdio.h>
int main(){
    int t; scanf("%d",&t);
    while(t--){
        int a,b,c;
        scanf("%d %d %d",&a,&b,&c);
        if((a+2*(b-a))>=c && (a+2*(b-a))%c==0 && (a+2*(b-a))!=0){printf("YES\n");}
        else if((2*b-c)>=a && (2*b-c)%a==0 && (2*b-c)!=0){printf("YES\n"); }
        else if((a+(c-a)/2)>=b && (c-a)%2==0 && (a+(c-a)/2)%b==0 && (a+(c-a)/2)!=0){printf("YES\n");  }
        else{printf("NO\n");}
    }
}
